hypervisor/arch/x86/guest/guest.c

```diff
@@ -327,22 +327,24 @@ static inline int32_t _copy_gpa(struct vm *vm, void *h_ptr, uint64_t gpa,
 static inline int copy_gpa(struct vm *vm, void *h_ptr, uint64_t gpa,
 	uint32_t size, bool cp_from_vm)
 {
-	int32_t len;
+	int32_t ret;
+	uint32_t len;
 
 	if (vm == NULL) {
 		pr_err("guest phy addr copy need vm param");
 		return -EINVAL;
 	}
 
 	do {
-		len = _copy_gpa(vm, h_ptr, gpa, size, 0, cp_from_vm);
-		if (len < 0)
-			return len;
+		ret = _copy_gpa(vm, h_ptr, gpa, size, 0, cp_from_vm);
+		if (ret < 0)
+			return ret;
 
+		len = (uint32_t) ret;
 		gpa += len;
```

Comment:
gpa is uint32_t, we need type casting here.

Reply:
gpa is uint64_t, len is uint32_t, and len will be promoted implicitly.
 Add cast not look well.